domain/application/types.go

```diff
@@ -383,6 +383,12 @@ type UnitAttributes struct {
 	ResolveMode string
 }
 
+type K8sPodInfo struct {
+	ProviderID string
```

Comment:
The providerID is of type `corenetwork.ID`

Reply:
Happy to do so now, it'll be a small change